src/main/dc-migration-assistant-fe/components/aws/quickstart/QuickStartDeploy.tsx

```diff
@@ -35,33 +43,79 @@ const QuickstartSubmitButton = styled(Button)`
     margin-top: 10px;
 `;
 
+const StackNameField = (): ReactElement => {
+    const fieldNameValidator = (stackName: string): string => {
+        const regExpMatch = stackName.match('^[a-zA-Z.-]+$');
+        return regExpMatch != null
+            ? undefined
+            : I18n.getText(
+                  'atlassian.migration.datacenter.provision.aws.form.stackName.validationMessage'
+              );
+    };
+
+    return (
+        <Field
+            validate={fieldNameValidator}
+            defaultValue=""
+            label={I18n.getText(
+                'atlassian.migration.datacenter.provision.aws.form.stackName.label'
+            )}
+            name="stackName"
+        >
+            {({ fieldProps, error }: any): ReactElement => (
+                <>
+                    <TextField width="medium" {...fieldProps} />
+                    {error && <ErrorMessage>{error}</ErrorMessage>}
+                </>
+            )}
+        </Field>
+    );
+};
+
 const QuickstartForm = ({
     quickstartParamGroups,
 }: Record<string, Array<QuickstartParameterGroup>>): ReactElement => (
     <Form
         onSubmit={(data: Record<string, any>): void => {
             const transformedCfnParams = data;
+            let stackNameValue = '';
```

Comment:
Because you've directly created the `TextField` now this is much simpler 😄
```suggestion
            const stackNameValue = transformedCfnParams['stackName']
            delete transformedCfnParams['stackName']
```